--- src/com/iota/iri/conf/Configuration.java
+++ src/com/iota/iri/conf/Configuration.java
@@ -102,7 +102,7 @@
         conf.put(DefaultConfSettings.REVALIDATE.name(), "false");
         conf.put(DefaultConfSettings.RESCAN_DB.name(), "false");
         conf.put(DefaultConfSettings.MAINNET_MWM.name(), "14");
-        conf.put(DefaultConfSettings.TESTNET_MWM.name(), "13");
+        conf.put(DefaultConfSettings.TESTNET_MWM.name(), "1");
 
         // Pick a number based on best performance
         conf.put(DefaultConfSettings.MIN_RANDOM_WALKS.name(), "5");
