--- src/com/iota/iri/Snapshot.java
+++ src/com/iota/iri/Snapshot.java
@@ -56,7 +56,6 @@
     private static void checkInitialSnapshotIsConsistent(Map<Hash, Long> initialState) {
         if (!isConsistent(initialState)) {
             log.error("Initial Snapshot inconsistent.");
-            System.exit(-1);
         }
     }
 
@@ -64,7 +63,6 @@
         long stateValue = initialState.values().stream().reduce(Math::addExact).orElse(Long.MAX_VALUE);
         if (stateValue != TransactionViewModel.SUPPLY) {
             log.error("Transaction resolves to incorrect ledger balance: {}", TransactionViewModel.SUPPLY - stateValue);
-            System.exit(-1);
         }
     }
 
@@ -88,7 +86,6 @@
             //syso is left until logback is fixed
             System.out.println("Failed to load snapshot.");
             log.error("Failed to load snapshot.", e);
-            System.exit(-1);
         }
         finally {
             IOUtils.closeQuietly(reader);
